pyecobee/__init__.py

```diff
@@ -271,13 +272,20 @@ def set_fan_mode(
                     "type": "setHold",
                     "params": {
                         "holdType": hold_type,
-                        "coolHoldTemp": int(cool_temp * 10),
-                        "heatHoldTemp": int(heat_temp * 10),
                         "fan": fan_mode,
                     },
                 }
             ],
         }
+
+        # Set optional parameters in body, only fill in *HoldTemp if they are non-zero
+        if cool_temp != 0:
+            body["functions"][0]["params"]["coolHoldTemp"] = int(cool_temp * 10)
+        if heat_temp != 0:
+            body["functions"][0]["params"]["heatHoldTemp"] = int(heat_temp * 10)
```

Comment:
Per my comment in the main thread, my view would be that we should change cool_temp and heat_temp to kwargs with a default value of None (assuming that it is possible to make the call to the ecobee API without including them???) and then adjust the behaviour downstream in Home Assistant to compensate for the change.